Authentication and new API URLs; Resolves #19

makbeta opened the pull request:
@RichardAtCT, I was able to make some updates to the code. I have done the following:
* Updated the API URLs in both OracleSwap and TransactionHelper classes
* Added authentication options to both OracleSwap and TransactionHelper classes
* Updated readme.md to include the added `api_key` parameter
* Added a bit more error details in OracleSwap
* Added a bit better error checking for None in TransactionHelper

I have done some testing, but not too extensively, didn't run any live transactions. However, the API's themselves didn't seem to have changed only the end points and authentication, so I relied on that. 

Looks like the gas oracle API is no longer there, I just added a comment but left the code is for a future update.

If some of the code is not up to par, don't hesitate to let me know. I hope these updates can be incorporated and be helpful to all.

RichardAtCT commented:
Thanks for the great pull request. My only concern before merging is that the transaction helper won't work without a valid gas-oracle. I have asked on 1inch discord if they have a option, otherwise will need to find a new source of gas price data.

makbeta commented:
@RichardAtCT I had an inkling that would be the case. Decided to submit the work so far, so there is progress on the issue. 
Let me know what the community says. 
I'll look around for other oracle options as well.

makbeta commented:
@RichardAtCT I tested the code and it was functional. I then more carefully tested that API, and it seems to be up and functional. Apologies for an inaccurate test and information the first time around.
Ex: https://gas-price-api.1inch.io/v1.3/1/
I think we should be good to go now on that account.

While testing, I also ran into an issue with passing of the address instead of a token name to a method, so I fixed it. I hope my fix is correct and acceptable. 
I also added a new chain that 1 Inch now supports as well. So the pull request is a bit bigger now, but I hope the improvements are welcome. But feel free to let me know if something needs to be revised.

RichardAtCT commented:
Well done on sorting out the gas! I’ll review during the course of today
and push to PiPy.

On Mon, 14 Aug 2023 at 7:45 AM makbeta ***@***.***> wrote:

> @RichardAtCT <https://github.com/RichardAtCT> I tested the code and it
> was functional. I then more carefully tested that API, and it seems to be
> up and functional. Apologies for an inaccurate test and information the
> first time around.
> Ex: https://gas-price-api.1inch.io/v1.3/1/
> I think we should be good to go now on that account.
>
> While testing, I also ran into an issue with passing of the address
> instead of a token name to a method, so I fixed it. I hope my fix is
> correct and acceptable.
> I also added a new chain that 1 Inch now supports as well. So the pull
> request is a bit bigger now, but I hope the improvements are welcome. But
> feel free to let me know if something needs to be revised.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RichardAtCT/1inch_wrapper/pull/20#issuecomment-1676627840>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHDKB3ZCCHTEKQG63CWIRX3XVGNNTANCNFSM6AAAAAA3ON5L3A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
